src/syntax_mapping.rs

```diff
@@ -251,10 +128,76 @@ impl<'a> SyntaxMapping<'a> {
 
 #[cfg(test)]
 mod tests {
+    use once_cell::sync::Lazy;
+    use rusty_fork::rusty_fork_test;
+
     use super::*;
     #[test]
-    fn basic() {
-        let mut map = SyntaxMapping::empty();
+    fn builtin_mappings_work() {
+        let map = SyntaxMapping::new();
+
+        assert_eq!(
+            map.get_syntax_for("/path/to/build"),
+            Some(MappingTarget::MapToUnknown)
+        );
+    }
+
+    #[test]
+    fn all_fixed_builtin_mappings_can_compile() {
+        let map = SyntaxMapping::new();
+
+        // collect call evaluates all lazy closures
+        // fixed builtin mappings will panic if they fail to compile
+        let _mappings = map.builtin_mappings().collect::<Vec<_>>();
+    }
+
+    // disabled for ARM builds because of an issue with `rusty_fork`
+    #[cfg(not(any(target_arch = "arm", target_arch = "aarch64")))]
+    // lazy initialisation test needs to be run on a separate instance because
+    // it will race with other tests
+    // see: https://github.com/rust-lang/rust/issues/47506
+    rusty_fork_test! {
```

Comment:
So, I looked in more detail at this test and  have some concerns. The first concern relates to the rusty-fork crate itself, which has not been updated in 3 years and thus seems unmaintained. That it does not work on arm also indicates that it is unmaintained.

Luckily I think we don't need rusty-fork, because I don't think we need this test. Don't get me wrong, I am a big fan of regression tests. But I am also a fan of long-term maintainability. And in my experience, one big factor in how to minimize long-term maintenance is to write tests that do not depend on implementations details.

This particular test is closely tied to implementation details, because it tests *how* fast startup is ensured in certain cases, rather than testing *that* fast startup is possible.

But, performance tests are tricky to run in a free CI where runners come in different capacities. The best mechanism we currently have to compare performance is tests/benchmarks/run-benchmarks.sh. It is not great, but IMHO it is better than tests that tests implementation details. So my preference would be to remove this test (and the rusty-fork depenency) and instead set up a test in tests/benchmarks/run-benchmarks.sh that looks something like this:

* measure the startup performance of the bat binary when run in a way where lazy mappings are not needed
* measure the startup performance of the bat binary when run in a way where all mappings are needed
* test that the bat binary starts faster in the first case than in the second case

This is a good test because it does not care *how* fast startup is achieved, only *that* it is achieved. We can completely change the implementation without changing the test at all.

I hope you don't find this comment discouraging! Keep up the good work :)

(I have not yet had time to look at any other code in the PR in detail)